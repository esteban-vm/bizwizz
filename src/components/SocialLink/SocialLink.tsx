import type { FC } from '@/types'
import { Icon, type IconName } from '../Icon'

type SocialLinkProps = {
  link: IconName
  style?: string
}

const SocialLink: FC<SocialLinkProps> = ({ link, style }) => {
  return (
    <a href={`https://www.${link}.com`} rel='noreferrer' target='_blank' title={link}>
      <Icon className={style} name={link} />
    </a>
  )
}

export default SocialLink
