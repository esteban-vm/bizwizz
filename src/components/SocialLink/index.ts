export { default as SocialLink, type SocialLinkProps } from './SocialLink'
