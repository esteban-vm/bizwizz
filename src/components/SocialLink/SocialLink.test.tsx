import { render, cleanup, within, screen } from '@/tests'
import SocialLink, { type SocialLinkProps } from './SocialLink'

describe('Social Link:', () => {
  let asFragment: () => DocumentFragment

  const props: SocialLinkProps = {
    id: '929262c9-7c62-4107-b17c-d1db9dbe3575',
    to: 'facebook',
    className: 'class',
  }

  beforeEach(() => {
    void ({ asFragment } = render(<SocialLink {...props} />))
  })

  afterEach(cleanup)

  it('should render correctly', () => {
    expect(asFragment()).toMatchSnapshot()
  })

  it('should have correct structure', () => {
    const linkElem = screen.getByRole('link')
    expect(linkElem).toBeInTheDocument()
    expect(linkElem).toHaveAccessibleDescription(props.to)
    expect(linkElem).toHaveAttribute('id', props.id)
    const iconElem = within(linkElem).getByRole('img')
    expect(iconElem).toBeInTheDocument()
    expect(iconElem).toHaveAttribute('class', props.className)
  })
})
