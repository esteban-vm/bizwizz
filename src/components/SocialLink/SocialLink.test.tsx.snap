// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Social Link: > should render correctly 1`] = `
<DocumentFragment>
  <a
    href="https://www.facebook.com"
    rel="noreferrer"
    target="_blank"
    title="facebook"
  >
    <svg
      role="img"
    >
      <use
        xlink:href="#icon-facebook"
      />
    </svg>
  </a>
</DocumentFragment>
`;
